### PR TITLE
Fix caly evo step slice

### PR DESCRIPTION
**Pull request by wangzyphysics**
## Summary by CodeRabbit

- **New Features**
  - Introduced new arguments and documentation for Calypso exploration configuration.
  - Enhanced trajectory parsing with additional filtering and selection criteria.

- **Bug Fixes**
  - Adjusted logic for handling `explore_style` parameter and mode assignments.

- **Refactor**
  - Renamed parameters and updated functions for consistency and clarity.
  - Removed unnecessary imports and adjusted variable names in tests.

- **Tests**
  - Expanded test cases to include new configurations and additional assertions.
  - Updated test methods to handle new trajectory configurations and verify outputs.

### CODE REVIEW

File: dpgen2/op/prep_caly_model_devi.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def get_input_sign(cls):
         return OPIOSign(
             {
                 "task_name": Parameter(str),
-                "template_slice_config": Parameter(dict),
+                "model_devi_group_size": Parameter(int),
```
Comment (wangzyphysics) [inline dpgen2/op/prep_caly_model_devi.py:51]: config

File: dpgen2/op/run_caly_model_devi.py [hunk 2]
```diff
@@ -272,6 +272,105 @@ def parse_traj(traj_file):
...
+        Atoms,
+    )
+    from ase.build import (  # type: ignore
+        make_supercell,
+    )
+    from ase.io import (  # type: ignore
+        read,
+    )
+
+    safe_dist_dict = {
```
[the remark this answers is not in the document]
Comment (wangzyphysics) [reply dpgen2/op/run_caly_model_devi.py:286]: config接口

File: dpgen2/op/prep_caly_model_devi.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ def get_input_sign(cls):
         return OPIOSign(
             {
                 "task_name": Parameter(str),
-                "template_slice_config": Parameter(dict),
+                "expl_config": BigParameter(dict),
```
Comment (wanghan-iapcm) [inline dpgen2/op/prep_caly_model_devi.py:51]: I would prefer naming the key by "config" to keep consistent with the input config of other calypso OPs

**Review by wanghan-iapcm — COMMENTED** [review]
I find the https://github.com/wangzyphysics/dpgen2/blob/3d0fba56af854abc1bd01540d8472d666bdd5fca/dpgen2/op/prep_caly_dp_optim.py#L58

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
Please check the following part. it seems that when "calypso:merge" and "calypso:default" are not supported. 

https://github.com/deepmodeling/dpgen2/blob/837e0259bd3ee499d0282e1c9856ab4d6b447c28/dpgen2/exploration/task/make_task_group_from_config.py#L510-L511

please raise an error if no condition is hit.

**Comment by wangzyphysics** [comment]
> Please check the following part. it seems that when "calypso:merge" and "calypso:default" are not supported.
> 
> https://github.com/deepmodeling/dpgen2/blob/837e0259bd3ee499d0282e1c9856ab4d6b447c28/dpgen2/exploration/task/make_task_group_from_config.py#L510-L511
> 
> please raise an error if no condition is hit.

This func `make_task_group_from_config` is not in use any more. 

```python
def make_naive_exploration_scheduler(
    config,
):
    # use npt task group
    explore_style = config["explore"]["type"]


    if explore_style == "lmp":
        return make_lmp_naive_exploration_scheduler(config)
    elif "calypso" in explore_style:
        return make_calypso_naive_exploration_scheduler(config)
    else:
        raise KeyError(
            f"Unknown key `{explore_style}`, Only support `lmp`, `calypso`, `calypso:merge` and `calypso:default`."
        )
```

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**